apps/vite/src/app.tsx

```diff
@@ -54,16 +74,37 @@ function App() {
           />
         </div>
         <main id="main">
-          <Topbar
-            title={`#${selectedChannel}`}
-            description="Share random funny stuff with the community. Play nice."
-            membersVisisble={showMembers}
-            onMembersPress={() => setShowMembers(show => !show)}
-          />
-          <div id="content">
+          <BlurView intensity={40} style={{ zIndex: 100 }}>
+            <Topbar
+              isBlurred={shouldBlurTopbar}
+              backgroundColor="$blurBackground"
+              icon={
+                icon &&
+                cloneElement(icon as ReactElement, { hasBackground: false })
+              }
+              title={`#${selectedChannel}`}
+              description="Share random funny stuff with the community. Play nice."
+              membersVisisble={showMembers}
+              onMembersPress={() => setShowMembers(show => !show)}
+            />
+          </BlurView>
+          <div id="content" ref={ref}>
             <Messages />
           </div>
-          <Composer />
+          <BlurView
+            intensity={40}
+            style={{
+              zIndex: 100,
+              marginTop: -112,
+              borderRadius: 20,
+            }}
+          >
+            <Composer
+              backgroundColor="$blurBackground"
+              paddingBottom={56}
+              isBlurred={shouldBlurBackground}
+            />
+          </BlurView>
```

Comment:
I was thinking of react-native use, but now that you talk about it we could use it directly inside the component. But i would have to try on native to see how the composer behaves. Because i think the use can be slightly different.